python/utils/worker.py

```diff
@@ -92,195 +92,258 @@ def run(self):
 
     def run(self):
         # Run DB migrations
-        print("[Parser] Initializing DB tables")
+
+        print(
+            json.dumps(
+                {
+                    "processor_name": self.processor.name(),
+                    "message": "[Parser] Initializing DB tables",
+                }
+            ),
+            flush=True,
+        )
         self.init_db_tables(self.processor.schema())
-        print("[Parser] DB tables initialized")
+        print(
+            json.dumps(
+                {
+                    "processor_name": self.processor.name(),
+                    "message": "[Parser] DB tables initialized",
+                }
+            )
+        )
 
         self.start_health_and_monitoring_ports()
 
         # Get starting version from DB
         starting_version = self.config.get_starting_version(self.processor.name())
         ending_version = self.config.ending_version
 
-        # Setup GRPC settings
-        metadata = (
-            ("x-aptos-data-authorization", self.config.grpc_data_stream_api_key),
-            ("x-aptos-request-name", self.processor.name()),
-        )
-        options = [
-            ("grpc.max_receive_message_length", -1),
-            (
-                "grpc.keepalive_time_ms",
-                self.config.indexer_grpc_http2_ping_interval_in_secs * 1000,
+        # Initialize the batch information for tracking
+        batch_start_version = starting_version
+        batch_end_version = None
+        prev_processed_start_version = None
+        prev_processed_end_version = None
```

Comment:
why do you need these?

Reply:
OK I kinda get prev_processed_end_version but not sure why we need the prev start version.

Reply:
Although...not sure why batch end version isn't enough. Isn't the current logic pull based on a batch_start_version, then process? You could just update batch_end_version at the end of the process and that's effectively your prev batch end version. 

If you switch to a separate thread handling the pulling this might change ofc.